yacht/lib/yacht.py

```diff
@@ -0,0 +1,121 @@
+import elasticsearch
+import os
+import datetime
+import time
+from esclientfactory import EsClientFactory
+from curatoresclientfactory import CuratorEsClientFactory
+from esutil import EsUtil
+from util import create_logger
+
+class IndexSource():
+
+    def __init__(self):
+        self.logger = create_logger(__name__,  os.getenv('YACHT_LOG_LEVEL', 'INFO'))
+        self.es = CuratorEsClientFactory.curator_es_client(False, 'ERROR')
+
+    def indices_tomorrow(self):
+        all_indices = self.get_indices()
+        filtered_indices = self.filter_indices(all_indices)
+        indices_tomorrow = self.format_indices(filtered_indices)
+        return indices_tomorrow
+
+    def get_indices(self):
+        all_indices = self.es.cat.indices('*', h='index,docs.count', format='json')
+        self.logger.debug('all indices: %s', all_indices)
+        return all_indices
+
+    def filter_indices(self, all_indices):
+        non_empty_indices = filter(lambda record: int(record['docs.count']) > 0 \
+            and self.index_name_allowed(record['index']), all_indices)
+        operateable_indices = map(lambda record: record['index'], non_empty_indices)
+        self.logger.debug('operateable_indices indices: %s', operateable_indices)
+        return operateable_indices
+
+    def format_indices(self, indices_today):
+        tomorrow = datetime.date.today() + datetime.timedelta(days=1)
+        indices_tomorrow = []
+        for index in indices_today:
+            items = index.split('.')
+            items[-1] = tomorrow.strftime('%d') # this way we get 'day' as a zero-padded string
+            items[-2] = tomorrow.strftime('%m')
+            items[-3] = tomorrow.strftime('%Y')
+            index_tomorrow = '.'.join(items)
+            indices_tomorrow.append(index_tomorrow)
+        self.logger.debug('indices_tomorrow indices: %s', indices_tomorrow)
+        return indices_tomorrow
+
+    def index_name_allowed(self, index_name):
+        if not index_name.startswith('project.'):
+            return False
+        subs = index_name.split('.')
+        parsed_date = datetime.date(int(subs[-3]), int(subs[-2]), int(subs[-1]))
+        if parsed_date != datetime.date.today():
+            return False
+        return True
+
+class IndexCreator():
+
+    def __init__(self):
+        self.logger = create_logger(__name__,  os.getenv('YACHT_LOG_LEVEL', 'DEBUG'))
+        self.wait_limit = int(os.getenv('MAX_WATING_TIME', '30000'))
+        self.retry_period = int(os.getenv('RETRY_PERIOD_SECONDS', '5'))
+        self.index_source = IndexSource()
+        self.es = CuratorEsClientFactory.curator_es_client(False, 'ERROR')
+        self.es_util = EsUtil(self.es)
+
+    def create(self):
+        self.es_util.server_running()
+        self.server_busy()
+        indices_tomorrow = self.index_source.indices_tomorrow()
+        for i in range(len(indices_tomorrow)):
```

Comment:
Consider using `enumerate`, might be a bit more pythonic.  Consider calculating length of indices once, as well.